source/Ship.cpp

```diff
@@ -2380,13 +2380,15 @@ bool Ship::Fire(vector<Projectile> &projectiles, vector<Visual> &visuals)
 
 
 // Fire an anti-missile.
-bool Ship::FireAntiMissile(const Projectile &projectile, vector<Visual> &visuals)
+bool Ship::FireAntiMissile(double distanceSquared, const Projectile &projectile, vector<Visual> &visuals)
 {
-	if(projectile.Position().Distance(position) > antiMissileRange)
+	if(distanceSquared > pow(antiMissileRange, 2))
 		return false;
 	if(CannotAct())
 		return false;
-	
+	if(projectile.IsDead())
+		return false;
+
```

Comment:
I am moving the check to Engine rather than removing it, not because I disagree with removing it, but just because I (as a brand new person bumbling around in the source purely for fun) do not feel it's my place to make that behavior change and would prefer to kick that decision to someone else.

Reply:
You can safely consider the request of a maintainer the "decision from someone else."